### ISSUE TITLE
Secretless has been XAd

### ISSUE THREAD

**Issue by izgeri** [cyberark/secretless-broker#643]
Key flows:
- [ ] Conjur provider in Kubernetes with PostgreSQL
- [ ] Conjur provider in Kubernetes with MySQL
- [ ] Conjur provider in OpenShift with PostgreSQL
- [ ] Conjur provider in OpenShift with MySQL

**Comment by izgeri** [comment]
All manual tests in the PR template should be run through.

**Comment by jonahx** [comment]
in the course XA, we found in the pg.yml file snippet in the kubernetes tutorial on the live site, the spacing is all borked atm.

**Comment by jonahx** [comment]
## Specific Bugs

- `$remote_appliance_url` is not explained [in this
  section](https://docs-staging.secretless.io/Latest/en/Content/Get%20Started/using-conjur.htm#CreatetheSecretlessBrokerconfiguration)
- Related to above point: Currently port 443 is hardoded, but it's not valid to
  assume the port will always be 443.  Need to update docs so both host and
  port are variables
- Just below that the line 
    ```
    create configmap conjur-ssl-cert --from-file=ssl-certificate="$(cat conjur.pem)"
    ```
  is wrong.  It should be:
    ```
    create configmap conjur-ssl-cert --from-file=ssl-certificate="$(cat conjur.pem)"
    ```
- Port forwaring is not explained anywhere, even though it's required to
  interact with OpenShift. Eg, we had to do:
  ```
  oc port-forward service/conjur-master 8443:443
  ```
- The section [Update Your
  Application](https://docs-staging.secretless.io/Latest/en/Content/Get%20Started/using-conjur.htm#UpdateYourApplication)
  is misleading bc it implies there is some task to do, but in fact it's merely
  explanatory.  We need to be consistent in the doc structure, and make sure
  required steps are clearly delineated from notes and context.

Also, you'll have to issue a new deployment.

- Setting up a host to authenticate via the Kubernetes Authenticator is not
  documented as part of the tutorial.  This includes application of policy. We
  just need to add the policy and explain what it's doing or have a seprate
  tutorial for that.  We had to do:
    ```
    ---
    - !policy
      id: my-service
      body:
      - &variables
        - !variable address
        - !variable username
        - !variable password

      - !group secrets-users

      - !host

      - !grant
        role: !group secrets-users
        member: !host

      - !permit
        resources: *variables
        privileges: [ read, execute ]
        role: !group secrets-users
    ```
- `secretless.yml` is broken...  messed up the configmap.
- "my-service" should be renamed because its a pod, not a k8s service
- "pg.yml" file is borked in k8s tutorial

## General comments

How to get postgres working in OpenShift? This was a rabbit hole. We had to use
a specific image, as well as port-forwarding.

The [Kubernetes
tutorial](https://secretless.io/docs/get_started/kubernetes_tutorial.html) is
k8s specific and assumes minikube. But this wasn't sufficient for OS.  Eg, we
had to use the internal DNS name of the service -- a networking detail we
don't discuss.  For external access, generally port-forwarding is the solution.

Perhaps add a note like: We're going to show you an example that uses
port-forwarding.

**How to login to Openshift**

```
oc login https://openshift-39.itci.conjur.net:8443/ --insecure-skip-tls-verify -u cadmin
```

**Comment by izgeri** [comment]
@jonahx I checked off the ones I believe I am addressing with cyberark/secretless-docs#128. I think I got all of the low-hanging fruit, with the exception of the configmap command (but I can't tell what's missing from that one).

**Comment by izgeri** [comment]
To address issues with the Conjur authn-k8s docs, we created cyberark/conjur-docs#166

**Comment by izgeri** [comment]
@jonahx can we close this issue? did we record everything that came out of XA in other issues as appropriate?

**Comment by jonahx** [comment]
Yes, we can close this.

We will not be doing an further XAing in the current state, as we've already identified problems.

Kumbi and I did 1 round of XA, and Srdjan and I did another (unfinished) round.  In both cases we found that Secretless worked as expected, but that there were serious problems due to the complexity setting up Conjur and the K8s authenticator.  It's likely that only a full step-by step, soup to nuts tutorial (ideally, with automation scripts to make it easier) will make the process tolerable or accessible to non-experts.
